optimade/models/structures.py

```diff
@@ -893,6 +895,29 @@ def check_anonymous_formula(cls, v):
 
         return v
 
+    @validator("chemical_formula_anonymous", "chemical_formula_reduced")
+    def check_reduced_formulae(cls, value, field):
+        if value is None:
+            return value
+
+        numbers = [n.strip() or 1 for n in re.split(r"[A-Z][a-z]*", value)]
```

Comment:
I guess you could still make this slightly shorter by using,
```suggestion
        numbers = [int(n.strip() or 1) for n in re.split(r"[A-Z][a-z]*", value)][1:]  
```
eliminating line 905

Although this is a bit of a mood point, because it would also make line 903 harder to read.

Reply:
I like the suggestion, but think I will leave it as it is so it will be easier to refactor and combine the validators in the future.